pkg/collector/corechecks/cluster/orchestrator/collector_bundle.go

```diff
@@ -281,12 +282,12 @@ func (cb *CollectorBundle) Run(sender sender.Sender) {
 
 		result, err := collector.Run(cb.runCfg)
 		if err != nil {
-			_ = cb.check.Warnf("Collector %s failed to run: %s", collector.Metadata().FullName(), err.Error())
+			_ = cb.check.Warnc(fmt.Sprintf("Collector %s failed to run: %s", collector.Metadata().FullName(), err.Error()), orchestrator.ExtraLogContext...)
 			continue
 		}
 
 		runDuration := time.Since(runStartTime)
-		log.Debugf("Collector %s run stats: listed=%d processed=%d messages=%d duration=%s", collector.Metadata().FullName(), result.ResourcesListed, result.ResourcesProcessed, len(result.Result.MetadataMessages), runDuration)
+		log.Debugc(fmt.Sprintf("Collector %s run stats: listed=%d processed=%d messages=%d duration=%s", collector.Metadata().FullName(), result.ResourcesListed, result.ResourcesProcessed, len(result.Result.MetadataMessages), runDuration), orchestrator.ExtraLogContext...)
```

Comment:
`Debugf` only performs the relatively expensive string formatting operation if the debug logs are enabled, while the new code will do it even if debug logs are disabled, wasting resources.

The same applies to all other `log.XXXc(fmt.Sprintf` calls, but debug logs are both more frequently disabled and larger in volume.

Is there a way to keep the formatting conditional on the log level?

Reply:
I added `xxxfc` in logs pkg to keep the formatting conditional on the log level
https://github.com/DataDog/datadog-agent/blob/45df03ed001d4904799fcc8e0012fd9180d332e6/pkg/util/log/log.go#L642-L644